src/Math/include/BipedalLocomotion/Math/Spline.h

```diff
@@ -0,0 +1,722 @@
+        return true;
+    }
+
+    if (t >= m_polynomials.back().finalPoint->time)
+    {
+        const auto finalTime = m_polynomials.back().finalPoint->time //
+                               - m_polynomials.back().initialPoint->time;
+        m_polynomials.back().getPosition(finalTime, position);
+        m_polynomials.back().getVelocity(finalTime, velocity);
+        m_polynomials.back().getAcceleration(finalTime, acceleration);
+        return true;
+    }
+
+    const auto poly = std::find_if(m_polynomials.begin(), m_polynomials.end(), [&t](const auto& p) {
+        return ((t >= p.initialPoint->time) && (t < p.finalPoint->time));
+    });
+
+    if (poly == m_polynomials.end())
+    {
+        log()->error("{} Unable to find the sub-trajectory.", logPrefix);
+        return false;
+    }
+
+    poly->getPosition(t - poly->initialPoint->time, position);
+    poly->getVelocity(t - poly->initialPoint->time, velocity);
+    poly->getAcceleration(t - poly->initialPoint->time, acceleration);
+
+    return true;
+}
+
+template <typename T>
+bool Spline<T>::evaluateOrderedPoints(const std::vector<std::chrono::nanoseconds>& t,
+                                      std::vector<T>& position,
+                                      std::vector<T>& velocity,
+                                      std::vector<T>& acceleration)
+{
+    constexpr auto logPrefix = "[Spline::evaluateOrderedPoints]";
+
+    if (!this->computeCoefficients())
+    {
+        log()->error("{} Unable to compute the coefficients of the spline.", logPrefix);
+        return false;
+    }
+
+    // resize the output vectors
+    position.resize(t.size());
+    velocity.resize(t.size());
+    acceleration.resize(t.size());
+
+    auto poly = m_polynomials.begin();
+
+    for (std::size_t i = 0; i < t.size(); i++)
+    {
+        // check if the time instant is before the first knot
+        if (t[i] < m_polynomials.front().initialPoint->time)
+        {
+            constexpr std::chrono::nanoseconds initialTime = std::chrono::nanoseconds::zero();
+            poly->getPosition(initialTime, position[i]);
+            poly->getVelocity(initialTime, velocity[i]);
+            poly->getAcceleration(initialTime, acceleration[i]);
+            continue;
+        }
+
+        // check if the time instant is after the last knot
+        if (t[i] >= m_polynomials.back().finalPoint->time)
+        {
+            const auto finalTime = m_polynomials.back().finalPoint->time //
+                                   - m_polynomials.back().initialPoint->time;
+            poly->getPosition(finalTime, position[i]);
+            poly->getVelocity(finalTime, velocity[i]);
+            poly->getAcceleration(finalTime, acceleration[i]);
+            continue;
+        }
+
+        if (poly != m_polynomials.end() && t[i] >= poly->finalPoint->time)
+        {
+            // advance the iterator
+            poly = std::find_if(poly, m_polynomials.end(), [&t, &i](const auto& p) {
```

Comment:
Nitpick
```suggestion
            poly = std::find_if(++poly, m_polynomials.end(), [&t, &i](const auto& p) {
```
In theory the condition would always fail on the first iteration. I don't think there is any gain anyhow :grin:

Reply:
Another possibility could also be to simply remove the ``if`` above.

Reply:
I applied you suggestion in https://github.com/ami-iit/bipedal-locomotion-framework/pull/773/commits/8e37f1e43a888c7cf6fc2bed833b59d8c5d47a37